src/ui/marker.js

```diff
@@ -765,7 +790,9 @@ export default class Marker extends Evented {
      * const alignment = marker.getRotationAlignment();
      */
     getRotationAlignment(): string {
-        return this._rotationAlignment === `auto` ? 'viewport' : this._rotationAlignment;
+        return (this._rotationAlignment === `auto` ||
+            (this._map && this._rotationAlignment === 'horizon' && !this._map._usingGlobe())) ?
+            'viewport' : this._rotationAlignment;
```

Comment:
Same suggestion as the other comment, this tries to do a bit much, to add some clarity we can expand as follow:

```suggestion
        if (this._rotationAlignment === `auto` || this._usingHorizonAlignment()) {
            return 'viewport';
        }
        return this._rotationAlignment;
```